### PR TITLE
core: rename `world` to `multiworld`

### PR DESCRIPTION
**Pull request by alwaysintreble**
discussed in the discord. renaming it to help prevent confusion for new devs. oot and smz3 broken currently

### CODE REVIEW

File: docs/world api.md [hunk 1]
```diff
@@ -446,8 +446,8 @@ In addition, the following methods can be implemented and attributes can be set
 
 ```python
 def generate_early(self) -> None:
-    # read player settings to world instance
-    self.final_boss_hp = self.world.final_boss_hp[self.player].value
+  # read player settings to world instance
+  self.final_boss_hp = self.multiworld.final_boss_hp[self.player].value
```
Comment (beauxq) [inline docs/world api.md:450]: We're trying to follow pep8 aren't we? Wouldn't we also want to follow it in the documentation?
Why the change to a 2-space indentation?

Comment (alwaysintreble) [reply docs/world api.md:450]: uhhh this definitely wasn't on purpose as i literally just ctrl+f for world...

**Comment by alwaysintreble** [comment]
took a minor stint where i was attempting to do python in Rider and it somehow completely butchered world api so that's fun.

**Review by ThePhar — APPROVED** [review]
Ran live-test game and several hundred test generations without issue.
